Modules/Challenges/test/Challenges.API.Tests.Integration/Models/RequestConfiguration.cs

```diff
@@ -1,21 +1,16 @@
-﻿using AutoMapper;
-using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.Mapping;
-
-namespace Challenges.API.Tests.Integration.Models;
-public class RequestConfiguration : IHaveCustomMapping
+﻿namespace Challenges.API.Tests.Integration.Models;
+public class RequestConfiguration
 {
     public AuthenticationParameters AuthenticationParameters { get; set; } = new AuthenticationParameters();
     public bool Authenticate { get; set; } = false;
     public string ContentType { get; set; } = string.Empty;
     public string AcceptHeader { get; set; } = string.Empty;
     public string Content { get; set; } = string.Empty;
 
-    public void CreateMappings(Profile configuration)
+    public void SupplementWith(RequestConfiguration other)
     {
-        configuration
-            .CreateMap<RequestConfiguration, RequestConfiguration>()
-            .ForMember(dest => dest.ContentType, opt => opt.Condition((src, dest) => string.IsNullOrEmpty(dest.ContentType)))
-            .ForMember(dest => dest.AcceptHeader, opt => opt.Condition((src, dest) => string.IsNullOrEmpty(dest.AcceptHeader)))
-            .ForMember(dest => dest.Content, opt => opt.Condition((src, dest) => string.IsNullOrEmpty(dest.Content)));
+        AcceptHeader = string.IsNullOrEmpty(AcceptHeader) ? other.AcceptHeader : AcceptHeader;
+        ContentType = string.IsNullOrEmpty(ContentType) ? other.ContentType : ContentType;
+        Content = string.IsNullOrEmpty(Content) ? other.Content : Content;
```

Comment:
That's not good. If you now set an AcceptHeader to, let's say "application/json" in the defaults, and in a test you say "Given an empty Accept header", then this results in an Accept header of "application/json", and you wouldn't even notice it in the tests. That's why I used null in my comment.

Reply:
Good call!